distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -1143,3 +1137,29 @@ private _local_file_move (source : File) (destination : File) (replace_existing
 private handle_invalid_path ~path ~action =
     Panic.catch InvalidPathException action caught_panic->
         Error.throw (Illegal_Argument.Error "The path "+path.to_display_text+" is invalid: "+caught_panic.payload.getMessage)
+
+## Interface for service providers willing to extend file system capabilities.
+   Register your implementation by adding following into your `package.yaml`:
+   ```yaml
+   services:
+   - provides: Standard.Base.System.File.File_System_SPI
+     with: <<fully qualified name of your implementation module>>
+   ```
+   In your module then define conversion of its associated type singleton
+   into `File_System_SPI` via its `new` static method.
+
+type File_System_SPI
+    private Entry protocol:Text typ
+
+    new protocol:Text typ = File_System_SPI.Entry protocol typ
+
+    private get_type protocol:Text b:Boolean =
+        all = Self.get_regs b
+        vec = all . filter \e -> (e.protocol == protocol)
+        if vec . not_empty then vec.first.typ else Nothing
+
+    private get_regs _:Boolean =
+        Meta_Helpers.lookup_services File_System_SPI
```

Comment:
The result of `get_regs` method may be:
```
[Entry "ssh" SSH_FS, Error.throw "Some error", Entry "smb" Sambal_FS]
```
Question, @GregoryTravis : How should this code and code in `get_type` be changed to report the error, but still work for other `ssh` and `smb` registrations?